settings.gradle

```diff
@@ -2,19 +2,60 @@ rootProject.name = 'xtext'
 include 'jenkins'
 include 'maven'
 
-['plugins', 'tests'].each { folder ->
+//collect all subprojects
+['plugins', 'tests','intellij', 'web'].each { folder ->
 	new File(settingsDir, folder).listFiles().findAll {
-		new File(it, 'build.properties').exists()
+		it.isDirectory() && it.name.startsWith("org.eclipse")
 	}
 	.each {
 		include "${folder}:${it.name}"
 	}
 }
 
-['intellij', 'web'].each {folder ->
-	new File(settingsDir, folder).listFiles().findAll {
-		it.isDirectory() && it.name.startsWith("org.eclipse")
-	}.each {
-		include "${folder}:${it.name}"
+/*
+ * We categorize all the projects and apply build logic to them accordingly
+ */
+gradle.beforeProject {project ->
+	project.with {
+		repositories {
+			jcenter()
+		}
+		apply from: "$rootDir/gradle/build-server-detection.gradle"
+		apply from: "$rootDir/gradle/eclipse-settings.gradle"
+		apply from: "$rootDir/gradle/java-compiler-settings.gradle"
+		apply from: "$rootDir/gradle/versions-from-parent-pom.gradle"
+		apply from: "$rootDir/gradle/versioning.gradle"
+		apply from: "$rootDir/gradle/dependency-hygiene.gradle"
+	
+		def isEclipsePlugin = project.file('build.properties').exists()
+		
+		if (isEclipsePlugin) {
```

Comment:
Personally I'd prefer clean entry points rather than the 'take them all and try to figure out later what they were' approach.